src/app/public/testing/dropdown/dropdown-fixture.ts

```diff
@@ -0,0 +1,187 @@
+ * of a component, such as changing its DOM structure.
+ */
+export class SkyDropdownFixture {
+
+  private debugEl: DebugElement;
+
+  /**
+   * Retrieves information about the dropdown component.
+   */
+  public get dropdown(): SkyDropdownFixtureDropdown {{
+    const button = this.buttonDebugElement;
+
+    if (!button) {
+      return undefined;
+    }
+
+    const buttonCmp = button.componentInstance as SkyDropdownComponent;
+
+    return {
+      buttonStyle: buttonCmp.buttonStyle,
+      buttonType: buttonCmp.buttonType,
+      disabled: buttonCmp.disabled,
+      dismissOnBlur: buttonCmp.dismissOnBlur,
+      horizontalAlignment: buttonCmp.horizontalAlignment,
+      label: buttonCmp.label,
+      title: buttonCmp.title
+    };
+  }}
+
+  /**
+   * Returns the dropdown button's text.
+   */
+  public get dropdownButtonText(): string {
+    return this.buttonDebugElement.nativeElement.innerText.trim();
+  }
+
+  /**
+   * Retrieves information about the dropdown menu component.
+   */
+  public get dropdownMenu(): SkyDropdownFixtureDropdownMenu {
+    const menu = this.getOverlay().querySelector('.sky-dropdown-menu');
+    if (!menu) {
+      return undefined;
+    }
+
+    return {
+      ariaLabelledBy: menu.getAttribute('aria-labelledby'),
+      ariaRole: menu.getAttribute('role')
+    };
+  }
+
+  /**
+   * Indicates if the dropdown menu is open and visible.
+   */
+  public get dropdownMenuIsVisible(): boolean {
+    const overlay = this.getOverlay();
+
+    if (!overlay) {
+      /* tslint:disable-next-line:no-null-keyword */
+      return false;
+    }
+
+    return this.getOverlay().querySelector('.sky-dropdown-menu') !== null;
+  }
+
+  private get buttonDebugElement(): DebugElement {
+    return this.debugEl.query(
+      By.css('.sky-dropdown-button')
+    );
+  }
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this.debugEl = SkyAppTestUtility.getDebugElementByTestId(fixture, skyTestId, 'sky-dropdown');
+  }
+
+  /**
+   * Click the dropdown button to open or close the dropdown menu.
+   */
+  public async clickDropdownButton(): Promise<any> {
+    this.buttonDebugElement.nativeElement.click();
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Click the dropdown item at the provided index.
+   */
+  public async clickDropdownItem(index: number): Promise<any> {
+    const itemEls = this.getDropdownItemEls();
+    if (index >= itemEls.length) {
+      throw new Error(`There is no dropdown item at index ${index}.`);
+    }
+
+    itemEls[index].click();
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Retrieves information about the dropdown item at the provided index.
+   */
+  public getDropdownItem(index: number): SkyDropdownFixtureDropdownItem {{
+    const item = this.getDropdownItemEls()[index];
+    if (!item) {
+      return undefined;
```

Comment:
A blank `return;` returns `undefined` by default (there are a few other occurences of this above).
```
return;
```

Reply:
Done.